packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java

```diff
@@ -164,19 +170,17 @@ public void onActivityPaused(Activity ignored) {
   public void onActivityResumed(Activity ignored) {
     if (isAuthSticky) {
       activityPaused = false;
-      final BiometricPrompt prompt =
-          new BiometricPrompt(activity, Executors.newSingleThreadExecutor(), this);
+      final BiometricPrompt prompt = new BiometricPrompt(activity, uiThreadExecutor, this);
       // When activity is resuming, we cannot show the prompt right away. We need to post it to the
       // UI queue.
-      new Handler(Looper.myLooper())
-          .postDelayed(
-              new Runnable() {
-                @Override
-                public void run() {
-                  prompt.authenticate(promptInfo);
-                }
-              },
-              100);
+      uiThreadExecutor.handler.postDelayed(
+          new Runnable() {
+            @Override
+            public void run() {
+              prompt.authenticate(promptInfo);
+            }
+          },
+          100);
```

Comment:
I realize the code was doing this already, but why the `postDelayed` with the 100ms delay here?

Reply:
See comment above:

```
// When activity is resuming, we cannot show the prompt right away. We need to post it to the
// UI queue.
```

Reply:
My confusion is around what's actually stopping us from showing the prompt immediately, and why 100ms specifically as the delay. Is it because we're waiting for `onResume` to totally finish, and we're hoping that `100ms` is enough time? There's an [onPostResume](https://developer.android.com/reference/android/app/Activity.html#onPostResume%28%29) callback that isn't used much but I think would work more consistently here. I'm not clear from the comment if it's `onResume` specifically that's the issue or if there's some other underlying state that we're waiting for.

Either way like I said, the code was doing this already--just noticed it now. I can file an issue to clean this up if needed.

Reply:
I don't like it either. We had to use it because onPostResume is only available after [Android Q](https://developer.android.com/reference/android/app/Application.ActivityLifecycleCallbacks.html#onActivityPostResumed(android.app.Activity)).

The 100ms delay? Completely random. I just tried a regular `post` instead of `postDelayed` and it worked. I will send a PR.